src/Text/Pandoc/Writers/ODT.hs

```diff
@@ -224,17 +224,36 @@ transformPicMath _ (Math t math) = do
          let dirname = "Formula-" ++ show (length entries) ++ "/"
          let fname = dirname ++ "content.xml"
          let entry = toEntry fname epochtime (fromStringLazy mathml)
-         modify $ \st -> st{ stEntries = entry : entries }
+         let fname' = dirname ++ "settings.xml"
+         let entry' = toEntry fname' epochtime $ isTextMode (t == InlineMath)
+         modify $ \st -> st{ stEntries = entry' : (entry : entries) }
          return $ RawInline (Format "opendocument") $ render Nothing $
-           inTags False "draw:frame" [("text:anchor-type",
-                                       if t == DisplayMath
-                                          then "paragraph"
-                                          else "as-char")
-                                     ,("style:vertical-pos", "middle")
-                                     ,("style:vertical-rel", "text")] $
+           inTags False "draw:frame" (if t == DisplayMath
+                                      then [("draw:style-name","fr2")
+                                            -- text:anchor-type="paragraph" does not work in LibreOffice 5.3
+                                           ,("text:anchor-type","as-char")]
+                                      else [("draw:style-name","fr1")
+                                           ,("text:anchor-type","as-char")]) $
              selfClosingTag "draw:object" [("xlink:href", dirname)
                                         , ("xlink:type", "simple")
                                         , ("xlink:show", "embed")
                                         , ("xlink:actuate", "onLoad")]
 
 transformPicMath _ x = return x
+
+isTextMode :: Bool -> B.ByteString
```

Comment:
Just as a stylistic matter, I'd expect a function named `isTextMode` to return a Bool.
Maybe `inTextMode` or `withTextMode` would be a better name?

Reply:
I renamed it to `documentSettings`. That's the name of the element it creates.